api/src/service/domain/workflow/workflowitem_create.ts

```diff
@@ -201,6 +207,42 @@ export async function createWorkflowitem(
     return new AlreadyExists(ctx, workflowitemCreated, workflowitemCreated.workflowitem.id);
   }
 
+  logger.trace({ item: workflowitemCreated }, "Check if assignee exists");
+  const userExistsResult = await repository.userExists(
+    workflowitemCreated.workflowitem.assignee
+  );
+
+  if (Result.isErr(userExistsResult)) {
+    return new VError(userExistsResult, "user exists check failed");
+  }
+
+  const userExists = userExistsResult;
+
+  if(!userExists) {
+    return new PreconditionError(
+      ctx,
+      workflowitemCreated,
+      "assigned user does not exist!");
+  }
+
+  const userResult = await repository.getUser(
+    workflowitemCreated.workflowitem.assignee
+    );
+
+  if (Result.isErr(userResult)) {
+    return new VError(userResult, "user check failed");
+  }
+
+  const userPermissions = userResult.permissions;
+
+  if(userPermissions["user.authenticate"] === undefined || userPermissions["user.authenticate"].length === 0) {
```

Comment:
the way you did it is fine but here's a short tipp how you can write cleaner code, you could refactor it this way:

`if(!userPermissions["user.authenticate"] || !userPermissions["user.authenticate"].length)`